src/pages/App.tsx

```diff
@@ -43,7 +43,7 @@ export class App extends React.Component<{}, State> {
         let dbcheck = timer(0, 100).subscribe(() => {
             getDbState().then((state) => {
                 this.setState({ ...this.state, dbState: state });
-                if (state.ready) {
+                if (state.ready || state.error != null) {
```

Comment:
checkForDbUpdate now populates the error property and returns the status if an issue occurred in updating the database, and will stop the UI from DOS attacking the middleware if it's never going to work.